src/app/core/utils/Result.ts

```diff
@@ -182,7 +186,7 @@ export function WrapResOr<T, E>(t: T | undefined, e: E): Result<T, E> {
     return t === undefined ? Err(e) : Ok(t);
 }
 export function WrapResOrE<T>(t: T | undefined, e: string): Result<T, MultiError> {
-    return WrapResOr(t, new MultiError([new Error(e)]));
+    return t === undefined ? Err(new MultiError([new Error(e)])) : Ok(t);
```

Comment:
Let's leave a comment here for posterity explaining why it's important to not construct Error here every time